tests/credential/conftest.py

```diff
@@ -39,21 +36,19 @@ def driver_and_saved_usernames(driver: Firefox, faker: Faker, origins):
     Yields a Tuple of the WebDriver object under test and a five-element list of the
     unique usernames added. (Five, because the last element was reused.)
     """
-    ba = BrowserActions(driver)
     faker.add_provider(internet)
     faker.add_provider(misc)
 
     def add_login(origin: str, username: str, password: str):
-        driver.get("about:logins")
-        about_logins = AboutLogins(driver)
+        about_logins = AboutLogins(driver).open()
         about_logins.add_login_button().click()
-        ba.clear_and_fill(about_logins.login_item_by_type("origin"), origin)
-        ba.clear_and_fill(about_logins.login_item_by_type("username"), username)
-        ba.clear_and_fill(about_logins.login_item_by_type("password"), password)
-        try:
-            about_logins.add_login_button()
-        except WebDriverException:
-            about_logins.login_item_save_changes_button().click()
+        about_logins.create_new_login(
+            {
+                "origin": origin,
+                "username": username,
+                "password": password,
+            }
+        )
```

Comment:
Not needed but `pytest-variables`

Reply:
Good call, will add the dep and convert to JSON files in the future to avoid Common SDET Problem number one zillion: Bunch of test data embedded in the tests.